treeherder/perf/management/commands/migrate_perf_data.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from optparse import make_option
+
+from concurrent.futures import ProcessPoolExecutor
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from treeherder.model.derived.jobs import JobsModel
+from treeherder.model.models import Datasource
+from treeherder.model.models import (MachinePlatform, OptionCollection,
+                                     Repository)
+from treeherder.perf.models import (PerformanceFramework,
+                                    PerformanceSignature,
+                                    PerformanceDatum)
+
+MAX_INTERVAL = 31536000
+
+
+def _migrate_series(project_name, signature_hash, signature_props):
+    framework = PerformanceFramework.objects.get(name='talos')
+    repository = Repository.objects.get(name=project_name)
+
+    with JobsModel(project_name) as jm:
+        if not PerformanceSignature.objects.filter(
+                signature_hash=signature_hash):
+            try:
+                option_collection = OptionCollection.objects.filter(
+                    option_collection_hash=signature_props['option_collection_hash'])[0]
+                platform = MachinePlatform.objects.filter(
+                    platform=signature_props['machine_platform'])[0]
+            except:
+                raise "Platform or object collection for %s (%s) does not exist" % (
+                    signature_hash, signature_props)
+
+            extra_properties = {}
+            for k in signature_props.keys():
+                if k not in ['option_collection_hash', 'machine_platform',
+                             'test', 'suite']:
+                    extra_properties[k] = signature_props[k]
+
+            signature, _ = PerformanceSignature.objects.get_or_create(
```

Comment:
You should use signature_hash to lookup an existing object and pass everything else to the `defaults` argument